### PR TITLE
CP2K - A100

### PR DESCRIPTION
**Pull request by RMeli**
Requires/contains #35.

### CODE REVIEW

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Comment by RMeli** [comment]
cscs-ci run cp2k

File: recipes/cp2k/a100/environments.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - cp2k@master +libxc +libint +spglib +cosma +sirius ~elpa build_system=cmake ^intel-oneapi-mkl ^cray-mpich
+  variants:
+    - +mpi
+    - +cuda
+    - cuda_arch=80
+    - build_type=Release
+  views:
+    dev:
+      exclude: ["cp2k"]
+    scalapack:
```
Comment (bcumming) [inline recipes/cp2k/a100/environments.yaml:23]: Should we call this view `cp2k-scalapack`? From the user's perspective, that is what the environment contains.
The same goes for the ELPA version below.

File: recipes/cp2k/a100/environments.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+tools:
+  compiler:
+  - toolchain: gcc
+    spec: gcc@11
+  unify: true
+  specs:
+  - cmake@3.25.2
+  - pkgconfig
+  - python@3.10
+  - py-numpy
```
Comment (bcumming) [inline recipes/cp2k/a100/environments.yaml:64]: Should we make a `tools` view?

File: recipes/cp2k/a100/modules.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        autoload: none
+      hash_length: 0
+      exclude_implicits: true
+      exclude: ['%gcc@7.5.0', 'gcc %gcc@7.5.0']
+      projections:
+        all: '{name}/{version}'
+        cp2k~elpa: 'cp2k/{version}'
+        sirius~elpa: 'sirius/{version}'
+        cp2k+elpa: 'cp2k+elpa/{version}'
+        sirius+elpa: 'sirius+elpa/{version}'
```
Comment (bcumming) [inline recipes/cp2k/a100/modules.yaml:25]: :+1:

File: stack-build [hunk 4]
```diff
@@ -19,7 +19,7 @@ get_build_id () {
 }
 
 usage () {
-    echo "usage: stack-build -n name -r recipe-path -s system -b build-root"
+    echo "usage: stack-build -n name -r recipe-path -s system -b build-root [-d]"
```
Comment (bcumming) [inline stack-build]: remove all of these `develop` changes, as they were merged in #32

**Review by bcumming — CHANGES_REQUESTED** [review]
Looking good!

I have a few suggestions, and one major suggestion: provide some documentation in the `/docs` path.
See `mkdocs.yaml` in the project root.

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Comment by RMeli** [comment]
cscs-ci run cp2k

File: docs/uenv-cp2k.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+* `cp2k-scalapack`: CP2K, dependencies, and [ScaLAPACK](https://www.netlib.org/scalapack/) as diagonalization library
+* `cp2k-scalapack-dev`: dependencies and [ScaLAPACK](https://www.netlib.org/scalapack/)
+* `cp2k-elpa`: CP2K, dependencies, and [ELPA](https://elpa.mpcdf.mpg.de/) as diagonalization library
+* `cp2k-elpa-dev`: dependencies and [ELPA](https://elpa.mpcdf.mpg.de/)
+
+## Building a custom version of CP2K
+
+### Using modules
+
+To build your version of QE do the following steps:
```
Comment (RMeli) [inline docs/uenv-cp2k.md:18]:
```suggestion
To build your version of CP2K do the following steps:
```

File: recipes/cp2k/a100/environments.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  unify: true
+  specs:
+  - cuda@11.8
+  - intel-oneapi-mkl +cluster
+  - fftw@3.3.10 +openmp
+  - libxc ~cuda
+  - sirius +fortran +scalapack
+  - dbcsr@develop +cuda cuda_arch=70 +shared +openmp ^intel-oneapi-mkl ^cray-mpich
+  - cosma@master +gpu_direct +scalapack +shared ^intel-oneapi-mkl ^cray-mpich
+  - cp2k@master +libxc +libint +spglib +cosma +sirius ~elpa build_system=cmake ^intel-oneapi-mkl ^cray-mpich
```
Comment (RMeli) [inline recipes/cp2k/a100/environments.yaml:14]:
```suggestion
  - cp2k@2023.2 +libxc +libint +spglib +cosma +sirius ~elpa build_system=cmake ^intel-oneapi-mkl ^cray-mpich
```

File: recipes/cp2k/a100/environments.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  - toolchain: gcc
+    spec: gcc@11
+  unify: true
+  specs:
+  - cuda@11.8
+  - intel-oneapi-mkl +cluster
+  - fftw@3.3.10 +openmp
+  - libxc ~cuda
+  - sirius +fortran +scalapack
+  - dbcsr@develop +cuda cuda_arch=70 +shared +openmp ^intel-oneapi-mkl ^cray-mpich
```
Comment (RMeli) [inline recipes/cp2k/a100/environments.yaml:12]:
```suggestion
  - dbcsr@2.6.0 +cuda +shared +openmp ^intel-oneapi-mkl ^cray-mpich
```

File: recipes/cp2k/a100/environments.yaml [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  - toolchain: gcc
+    spec: gcc@11
+  unify: true
+  specs:
+  - cuda@11.8
+  - intel-oneapi-mkl +cluster
+  - fftw@3.3.10 +openmp
+  - libxc ~cuda
+  - sirius +fortran + scalapack +elpa
+  - dbcsr@develop +cuda cuda_arch=70 +shared +openmp ^intel-oneapi-mkl ^cray-mpich
```
Comment (RMeli) [inline recipes/cp2k/a100/environments.yaml:39]:
```suggestion
  - dbcsr@2.6.0 +cuda +shared +openmp ^intel-oneapi-mkl ^cray-mpich
```

File: recipes/cp2k/a100/environments.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  - cosma@master +gpu_direct +scalapack +shared ^intel-oneapi-mkl ^cray-mpich
+  - cp2k@master +libxc +libint +spglib +cosma +sirius +elpa build_system=cmake ^intel-oneapi-mkl ^cray-mpich
```
Comment (RMeli) [inline recipes/cp2k/a100/environments.yaml:41]:
```suggestion
  - cp2k@2023.2 +libxc +libint +spglib +cosma +sirius +elpa build_system=cmake ^intel-oneapi-mkl ^cray-mpich
```

**Review by RMeli — COMMENTED** [review]
Pin version of CP2K to `2023.2` and DBCSR to `2.6.0`.

`cuda_arch=70` for DBCSR is no longer required since https://github.com/spack/spack/pull/38939.

**Comment by RMeli** [comment]
Waiting for https://github.com/spack/spack/pull/39782.

**Comment by RMeli** [comment]
cscs-ci run cp2k

**Review by bcumming — APPROVED** [review]

**Pull request merged**